app/src/main/java/com/mk/tasky/agenda/data/mapper/ReminderMapper.kt

```diff
@@ -0,0 +1,35 @@
+package com.mk.tasky.agenda.data.mapper
+
+import com.mk.tasky.agenda.data.local.entity.ReminderEntity
+import com.mk.tasky.agenda.domain.model.Reminder
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.util.*
+
+fun Reminder.toEntity(): ReminderEntity {
+    return ReminderEntity(
+        id = this.id ?: UUID.randomUUID().toString(),
+        title = this.title,
+        description = this.description,
+        remindAt = this.remindAt.atZone(ZoneId.systemDefault()).toInstant().toEpochMilli(),
+        day = this.time.dayOfMonth,
+        month = this.time.monthValue,
+        year = this.time.year,
+        hour = this.time.hour,
+        minute = this.time.minute
+    )
+}
+
+fun ReminderEntity.toDomain(): Reminder {
+    return Reminder(
+        id = this.id,
+        title = this.title,
+        description = this.description,
+        remindAt = LocalDateTime.ofInstant(
+            Instant.ofEpochMilli(this.remindAt),
+            ZoneId.systemDefault()
+        ),
+        time = LocalDateTime.of(year, month, day, hour, minute)
```

Comment:
Here at this point, the timezone conversion issue would happen.
Imagine the user saves a reminder in America time, travels and then opens the app from Europe - these values here wouldn't change, but the actual time does.

Reply:
You are right. Fixed!